### DISCUSSION TITLE
Short waveforms rendering differently in v7 vs. v6

### DISCUSSION THREAD

**Discussion by giilby** [katspaugh/wavesurfer.js#3885] [Q&A]
After upgrading from v6.6.4 to v7.8.6, I noticed that waveforms for short audio clips are rendering much differently when in a wide container. To start, here's a comparison of old vs. new for narrow containers:

v6:
![v6-narrow](https://github.com/user-attachments/assets/7fe05ca0-a302-434c-92b7-9972e52f8e11)

v7:
![v7-narrow](https://github.com/user-attachments/assets/07024df3-98c6-4c0f-b2d0-d6967ac562c1)

When the same clip is rendered in a wider container (around 1800px):

v6:
![v6-wide](https://github.com/user-attachments/assets/4e30f984-d252-40e2-b80c-971600ced125)

v7:
![v7-wide](https://github.com/user-attachments/assets/16cc914c-d06e-460c-95b9-4628ed6a7819)

The audio clip is 5 seconds long, and I've configured WaveSurfer with these size-related properties:

```
{
  cursorWidth: 0,
  barWidth: 1,
  barGap: 1,
  height: 40,
}
```

It seems like the old renderer was repeating bars to respect `barWidth` and `barGap` in scenarios like this, but maybe something else is going on. I reviewed the config options but didn't see anything else that seemed to be related. Any idea what's going on here? Any additional information I can provide?

Thanks!

**Comment by katspaugh** [comment]
For short clips you might want to increase the sample rate to get more peaks to render (e.g. pass `sampleRate: 44100` in the options).
By default, sample rate is 8000 in v7 as a performance optimization.

**Reply by giilby** [reply]
I'm using peak data JSON files, so changing the sample rate wouldn't make a difference, right? Seems like v6 was interpolating/repeating data in this case, but given the same exact JSON file, v7 is only drawing exactly what's given. Any workaround you can see? Thanks for your help, and for building a great library!

**Reply by katspaugh** [reply]
Gotcha, I’ll make an issue for this.

**Reply by giilby** [reply]
Awesome. Thanks!
